docs/04-refactor-state-management.md

```diff
@@ -91,126 +90,125 @@ public void ResetOrder()
 {
     Order = new Order();
 }
+
+public void RemoveConfiguredPizza(Pizza pizza)
+{
+    Order.Pizzas.Remove(pizza);
+}
 ```
 
-## Updating ConfigurePizzaDialog
+## Exploring state changes
 
-As a next step, let's give the `ConfigurePizzaDialog` the same treatment.
+At this point it should be possible to get the `Index` component compiling again by updating references to refer to various bits attached to `OrderState`. Feel free to create convenience properties for things like `OrderState.Order` or `OrderState.Order.Pizzas` if it feels better to you that way.
 
-First inject `OrderState` into the `ConfigurePizzaDialog` by adding `@inject OrderState OrderState` at the top. Then you can remove the parameter properties (i.e., the ones with the [Parameter] attribute). We no longer need the parameters because `ConfigurePizzaDialog` will get its data from `OrderState` instead. 
+Try this out and verify that everything still works.
 
-Also make sure to go back to `Index` and remove the arguments to the `<ConfigurePizzaDialog />`. If you see then exception message like "Component ConfigurePizzaDialog does not have a parameter named 'Pizza'." then this is the problem.
+This is a good opportunity to explore how state changes and rendering work in Blazor, and how `EventCallback` solves some common problems. The detail of what are happening now became more complicated now that `OrderState` involved.
```

Comment:
Topic for discussion - the workshop content no longer explains `StateHasChanged()` in the exercises because you never need to do it. The auth state provider uses is, but nothing you write during the workshop will teach you about that. If we wanted to include this we could consider adding a section where you write a cascading value provider.